src/index.ts

```diff
@@ -0,0 +1,188 @@
+        // the same name as one previously registered. So, we tell them about it.
+        if (printFragmentWarnings) {
+          console.warn("Warning: fragment with name " + fragmentName + " already exists.\n"
+            + "graphql-tag enforces all fragment names across your application to be unique; read more about\n"
+            + "this in the docs: http://dev.apollodata.com/core/fragments.html#unique-names");
+        }
+
+        fragmentSourceMap[fragmentName][sourceKey] = true;
+
+      } else if (!fragmentSourceMap.hasOwnProperty(fragmentName)) {
+        fragmentSourceMap[fragmentName] = {};
+        fragmentSourceMap[fragmentName][sourceKey] = true;
+      }
+
+      if (!astFragmentMap[sourceKey]) {
+        astFragmentMap[sourceKey] = true;
+        definitions.push(fragmentDefinition);
+      }
+    } else {
+      definitions.push(fragmentDefinition);
+    }
+  }
+
+  // TODO: Ts won't allow us to alter this since it's read-only
+  (ast as any).definitions = definitions;
+  return ast;
+}
+
+export function disableFragmentWarnings(): void {
+  printFragmentWarnings = false;
+}
+
+function stripLoc(doc: Array<DocumentNode> | DocumentNode, removeLocAtThisLevel: number | boolean): DocumentNode  {
+  const docType = Object.prototype.toString.call(doc);
+
+  if (docType === '[object Array]' && doc instanceof Array) {
+    // This is wrong now because we return a DocumentNode... How do we approach this?
+    return doc.map((d: DocumentNode) => stripLoc(d, removeLocAtThisLevel)) as any;
+  }
+
+  if (docType !== '[object Object]') {
+    throw new Error('Unexpected input.');
+  }
+
+  doc = doc as DocumentNode;
+
+  // We don't want to remove the root loc field so we can use it
+  // for fragment substitution (see below)
+  if (removeLocAtThisLevel && doc.loc) {
+    // TODO: TS Does not allow deleting a read-only prop
+    delete (doc as any).loc;
+  }
+
+  // https://github.com/apollographql/graphql-tag/issues/40
+  if (doc.loc) {
+    // TODO: TS Does not allow deleting a read-only prop
+    delete (doc as any).loc.startToken;
+    delete (doc as any).loc.endToken;
+  }
+
+  const keys = Object.keys(doc);
+  let key;
+  let value;
+  let valueType;
+
+  for (key in keys) {
+    if (keys.hasOwnProperty(key)) {
+      // TODO: type this, need some more insight
+      value = (doc as any)[keys[key]];
+      valueType = Object.prototype.toString.call(value);
+
+      if (valueType === '[object Object]' || valueType === '[object Array]') {
+        // TODO: type this, need some more insight
+        (doc as any)[keys[key]] = stripLoc(value, true);
+      }
+    }
+  }
+
+  return doc;
+}
+
+let experimentalFragmentconstiables = false;
+function parseDocument(doc: string): DocumentNode {
+  const cacheKey = normalize(doc);
+
+  if (docCache[cacheKey]) {
+    return docCache[cacheKey];
+  }
+
+  // TODO: TS asked to change this from "experimentalFragmentconstiables" to "experimentalFragmentVariables"
```

Comment:
i think your conversion from `var` to `const` was too greedy. making the adjustment above should resolve this